Added Backend and Signin  and Signup

NOOB-3301 opened the pull request:
## Fixing #45 

### Changes:

1. Added backend setup
2. added login and register controllers
3. It also store the jwt token in localstore

### Steps to setup:
1. i created a server folder where all server backend resides.
2. install required packages by `npm i`
4. create a `.env` at root of the project
5. add your mongodb url there `MONGO_URI="mongodb url"`
6. after that start your server
7. The routes are-
    Base Route- `/api/auth`
    * signin route `/login`
    * signup route `/signup`

saismrutiranjan18 commented:
> ## Fixing #45
> ### Changes:
> 1. Added backend setup
> 2. added login and register controllers
> 3. It also store the jwt token in localstore
> 
> ### Steps to setup:
> 1. i created a server folder where all server backend resides.
> 2. install required packages by `npm i`
> 3. create a `.env` at root of the project
> 4. add your mongodb url there `MONGO_URI="mongodb url"`
> 5. after that start your server
> 6. The routes are-
>    Base Route- `/api/auth`
>    
>    * signin route `/login`
>    * signup route `/signup`

Will you provide any screen recording, and something like this , so i am able to understand how it works @NOOB-3301 
![image](https://github.com/user-attachments/assets/a629ec14-2fa5-446a-94fc-ed7707d041b7)
It show this, @NOOB-3301

NOOB-3301 commented:
> > ## Fixing #45
> > ### Changes:
> > 
> > 1. Added backend setup
> > 2. added login and register controllers
> > 3. It also store the jwt token in localstore
> > 
> > ### Steps to setup:
> > 
> > 1. i created a server folder where all server backend resides.
> > 2. install required packages by `npm i`
> > 3. create a `.env` at root of the project
> > 4. add your mongodb url there `MONGO_URI="mongodb url"`
> > 5. after that start your server
> > 6. The routes are-
> >    Base Route- `/api/auth`
> >    
> >    * signin route `/login`
> >    * signup route `/signup`
> 
> Will you provide any screen recording or something like this , so i am able to understand how it works @NOOB-3301 ![image](https://private-user-images.githubusercontent.com/160756794/400991161-a629ec14-2fa5-446a-94fc-ed7707d041b7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzYzMTU1NzgsIm5iZiI6MTczNjMxNTI3OCwicGF0aCI6Ii8xNjA3NTY3OTQvNDAwOTkxMTYxLWE2MjllYzE0LTJmYTUtNDQ2YS05NGZjLWVkNzcwN2QwNDFiNy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMTA4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDEwOFQwNTQ3NThaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zM2Q2OWJkMWY2YjIyMTNiNTE5NzI0OGY4NGUyMzUzNGU4MWMzZjk5YWRiMzlhODkzNGEyMjNmZmJmMTdjNzE0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.vb2GaVh3p-PRHaBaWPoSfLIYeGFofiNoYjaL1qrOBnk) It show this, @NOOB-3301

[Drive link](https://drive.google.com/file/d/1Lm5QWwv9fkV8wdYJj5pRJKTUboFjQWNr/view?usp=sharing)

Now as the backend add you have to host the backend seerver to a different vercel project. on vercel website when deploying your project set the server as your root directory.
![image](https://github.com/user-attachments/assets/04df7086-e6b3-4242-8ffd-18dec72b8343)

and after deploying set the  url in `Baselink.js`  to the vercel deployed url from localhost

NOOB-3301 commented:
I have dm u in discord have a look @saismrutiranjan18 
As the pr is merged kindly assign me the points for this pr. link https://os-lead.vercel.app/
